feature_engine/selection/target_mean_encoding.py

```diff
@@ -0,0 +1,400 @@
+from sklearn.base import TransformerMixin, BaseEstimator
+from sklearn.utils.validation import check_is_fitted
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import roc_auc_score, r2_score
+
+from feature_engine.variable_manipulation import _find_all_variables, _define_variables
+from feature_engine.dataframe_checks import _is_dataframe
+
+import pandas as pd 
+
+class TargetMeanEncoding(BaseEstimator, TransformerMixin):
+    """
+    TargetMeanEncoding
+    ------------------
+        Description
+        -----------
+        Calculates the feature importance.
+
+        For each categorical variable:
+            1) Separate into train and test
+            2) Determine the mean value of the target within each label of the categorical variable using the train set
+            3) Use that mean target value per label as the prediction (using the test set) and calculate the roc-auc.
+
+        For each numerical variable:
+            1) Separate into train and test
+            2) Divide the variable into 100 quantiles
+            3) Calculate the mean target within each quantile using the training set 
+            4) Use that mean target value / bin as the prediction (using the test set) and calculate the roc-auc
+        
+        Implementation
+        --------------
+
+            Private methods
+            ---------------
+                `__target_mean_encoding(self, df_train_, df_test_, var_cols=None)`
+                `__validate_variables(self, X)`
+                    
+            Public methods
+            --------------
+                `fit(self, X, y=None)`
+                `transform(self)`
+                `fit_transform(self, X, y=None)`
+        
+    Parameters
+    ----------
+
+    variables: list, default=None
+        The list of variables to evaluate. If None, the transformer will evaluate all 
+        variables in the dataset associated with the variables_type.
+
+    target: string, default=None
+        The target variable to evaluate. If None, the transformer will return the 
+        relevant message for the missing value.
+
+    variables_type: boolean, default=True
+        If variables_type = True: the transformer will evaluate the variables as the 
```

Comment:
I don't think we need this parameter.  _define_all_variables in fit will select all variables. If the user wants a specific set, they should input it manually. This way this transformer follows the same logic and use of all the remaning transformers from sklearn. Could you please remove?

Reply:
Hi Sole,
I still not figure out what it would be if we don't have this parameter. What scenarios should be covered here? I could have several cases as follows.
1. No variables at input: the output will be the categorical variables for feature selection.
2. With variables: 
a. check if the inputs are numerical variables then calculate using the numerical process, otherwise, calculate them using the categorical process. 
b. check if the inputs are mixed, then calculate the default mode using the categorical process.
Is it okay to do this approach? Please let me know your thought. Thanks Sole!

Reply:
pandas can detect the type of the variables. So we should have this functionality as part of the transformer and not ask the user to enter it. Thus, we do not need this parameter. We will do the variable type identification as part of the process.

In fact, if we use a combination of the transformers I suggested, they should automatically find numerical and categorical variables. So I am thinking that we do not need to add this functionality at all for this transformer, as it is already part of the other 2